tests/aws/services/lambda_/test_lambda_api.py

```diff
@@ -3998,6 +3999,49 @@ def test_url_config_lifecycle(self, create_lambda_function, snapshot, aws_client
             aws_client.lambda_.get_function_url_config(FunctionName=function_name)
         snapshot.match("failed_getter", ex.value.response)
 
+    @markers.aws.only_localstack
+    def test_create_url_config_custom_id_tag(self, create_lambda_function, aws_client):
+        custom_id_value = "my-custom-subdomain"
+
+        function_name = f"test-function-{short_uid()}"
+        create_lambda_function(
+            func_name=function_name,
+            zip_file=testutil.create_zip_file(TEST_LAMBDA_NODEJS, get_content=True),
+            runtime=Runtime.nodejs20_x,
+            handler="lambda_handler.handler",
+            Tags={TAG_KEY_CUSTOM_URL: custom_id_value},
+        )
+        url_config_created = aws_client.lambda_.create_function_url_config(
+            FunctionName=function_name,
+            AuthType="NONE",
+        )
+        # Since we're not comparing the entire string, this should be robust to
+        # region changes, https vs http, etc
+        assert f"://{custom_id_value}.lambda-url." in url_config_created["FunctionUrl"]
+
+    @markers.aws.only_localstack
+    def test_create_url_config_custom_id_tag_invalid_id(
+        self, create_lambda_function, aws_client, caplog
+    ):
+        custom_id_value = "_not_valid_subdomain"
+
+        function_name = f"test-function-{short_uid()}"
+        create_lambda_function(
+            func_name=function_name,
+            zip_file=testutil.create_zip_file(TEST_LAMBDA_NODEJS, get_content=True),
+            runtime=Runtime.nodejs20_x,
+            handler="lambda_handler.handler",
+            Tags={TAG_KEY_CUSTOM_URL: custom_id_value},
+        )
+
+        caplog.clear()
+        url_config_created = aws_client.lambda_.create_function_url_config(
+            FunctionName=function_name,
+            AuthType="NONE",
+        )
+        assert any("Invalid custom ID tag value" in record.message for record in caplog.records)
```

Comment:
question: since I never used `caplog` before, is it possible that for the log level in CI to be lower than the logged statement here, and it would not show up? Just trying to think if we need to manually set the log level with `caplog.set_level(logging.WARNING)` to avoid any breakage.

@dominikschubert any opinion on this? I see `caplog` was used in lambda before, but as I have never used it, I'm trusting you and Nick to confirm it to me 😅

Reply:
In theory, yes, it's possible, but it would require someone to manually set the loglevel for the root logger (or the loglevel for the specific logger... python logging is a hot mess) to something above ``WARNING``. That would cause all warnings in the entirety of localstack to be ignored, at which point, I think more things than just this assertion would break!

Also, keep in mind that ``caplog.set_level(logging.WARNING)`` would then swallow anything lower than a warning. *If* someone wanted to go that route, they'd also want to compare it with the existing loglevel, and set whichever one was greater, warning or the current one.

Reply:
Okay, thanks! I'm not too sure I follow:

We could theoretically lower the global log output in CI for some reason, I know we've reduced it lately from `DEBUG` to `WARNING` I think.
I believe we don't generally assert on logs in our tests, so I don't think setting the log level to `ERROR` would break anything currently.
Which is why I was thinking of setting the log level manually with `caplog.set_level` for this test, so that the conditions for the test are always met, and don't depend on external settings? I see also the point with comparing with current log level, in order to not reduce the verbosity. 
It seems this would only set the log level for this test, and be reset after. From the [`pytest` documentation ](https://docs.pytest.org/en/7.1.x/how-to/logging.html):
> The log levels set are restored automatically at the end of the test.

To me, in order to make the test solid, manually setting the log level to the desired level would reduce the possibility of external changes breaking it, what do you think? I mean, it's not a blocker, just a thought, as it's not a pattern we're seeing often, and I was curious about it 😄

Reply:
Would generally agree with the reasoning presented by Ben. It's not a common pattern we're using to assert on logs and if used, I'd agree that we should at least make sure it's very clear why a test would now fail, e.g. by asserting on the debug level or as @bentsku suggest, manually setting it to the required level for the test duration. :thinking: